external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/KafkaSpout.java

```diff
@@ -330,11 +328,9 @@ public void ack(Object messageId) {
     public void fail(Object messageId) {
         final KafkaSpoutMessageId msgId = (KafkaSpoutMessageId) messageId;
         emitted.remove(msgId);
-        if (msgId.numFails() < maxRetries) {
-            msgId.incrementNumFails();
-            retryService.schedule(msgId);
-        } else { // limit to max number of retries
-            LOG.debug("Reached maximum number of retries. Message [{}] being marked as acked.", msgId);
+        msgId.incrementNumFails();
+        if (!retryService.schedule(msgId)) {
+            LOG.debug("Retry service indicated message should not be retried. Message [{}] being marked as acked.", msgId);
```

Comment:
I would leave the message as it was before. "Reached maximum number of retries. Message [{}] being marked as acked."

It adds overhead and it is irrelevant to be mentioning the retry service here.